library/job.go

```diff
@@ -50,44 +51,112 @@ var (
 	errNotJobID = errors.NewKind("couldn't assign an ID to a job")
 )
 
+// NewDownloadJobScheduleFn builds a new gitcollector.ScheduleFn that only
+// schedules download jobs.
+func NewDownloadJobScheduleFn(
+	lib borges.Library,
+	download chan gitcollector.Job,
+	updateOnDownload bool,
+	jobLogger log.Logger,
+	temp billy.Filesystem,
+) gitcollector.ScheduleFn {
+	return func(
+		opts *gitcollector.JobSchedulerOpts,
+	) (gitcollector.Job, error) {
+		job, err := jobFrom(download, opts.JobTimeout)
+		if err != nil {
+			return nil, err
+		}
+
+		job.Lib = lib
+		job.TempFS = temp
+		job.Update = updateOnDownload
+		job.Logger = jobLogger
+		return job, nil
+	}
+}
+
+// NewUpdateJobScheduleFn builds a new gitcollector.SchedulerFn that only
+// schedules update jobs.
+func NewUpdateJobScheduleFn(
+	lib borges.Library,
+	update chan gitcollector.Job,
+	jobLogger log.Logger,
+) gitcollector.ScheduleFn {
+	return func(
+		opts *gitcollector.JobSchedulerOpts,
+	) (gitcollector.Job, error) {
+		job, err := jobFrom(update, opts.JobTimeout)
+		if err != nil {
+			return nil, err
+		}
+
+		job.Lib = lib
+		job.Logger = jobLogger
+		return job, nil
+	}
+}
+
 // NewJobScheduleFn builds a new gitcollector.ScheduleFn that schedules download
 // and update jobs in different queues.
 func NewJobScheduleFn(
 	lib borges.Library,
 	download,
 	update chan gitcollector.Job,
+	updateOnDownload bool,
 	jobLogger log.Logger,
 	temp billy.Filesystem,
 ) gitcollector.ScheduleFn {
+	var (
+		downloadClosed bool
+		updateClosed   bool
+	)
+
 	return func(
 		opts *gitcollector.JobSchedulerOpts,
 	) (gitcollector.Job, error) {
-		if len(download) == 0 && len(update) == 0 {
-			return nil, gitcollector.ErrNewJobsNotFound.New()
-		}
-
 		var (
 			job *Job
 			err error
 		)
 
-		if len(download) > 0 {
-			job, err = jobFrom(download, opts.JobTimeout)
-		} else {
+		job, err = jobFrom(download, opts.JobTimeout)
```

Comment:
If I am not mistaken if there's no download job it will wait the timeout before checking if there's an update job. Maybe we can keep it like this now but I would open an issue to change the method to check if there's a job with a select.

Reply:
Issue opened at #18